### PR TITLE
gCMT Catalog accessor

### PR DESCRIPTION
**Pull request by pabloitu**
Added gcmt accessor from ISC API.  Follows the same logic than comcat and bsi readers. Created test for an event search and summary info from a request.

# pyCSEP Pull Request Checklist

Please check out the [contributing guidelines](https://github.com/SCECcode/pycsep/blob/master/CONTRIBUTING.md) for some tips 
on making pull requests to pyCSEP. 

Fixes issue #(*please fill in or delete if not needed*).

## Type of change:

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: csep/utils/readers.py [hunk 1]
```diff
@@ -702,44 +719,182 @@ def _query_bsi(start_time, end_time, min_magnitude=2.50,
...
-                       starttime=start_time, endtime=end_time, **extra_bsi_params)
+                       starttime=start_time, endtime=end_time,
+                       **extra_bsi_params)
 
     return eventlist
-def _parse_datetime_to_zmap(date, time):
-        """ Helping function to return datetime in zmap format.
+
+
+def _query_gcmt(out_format='QuakeML',
```
Comment (fabiolsilva) [inline csep/utils/readers.py:728]: I think this is now conflicting with the just merged pull request for the GNS catalog. This function is exactly at the same place as query_gns, and is causing the conflict. Could you resolve the conflict using the updated version in the master branch? Once that is done, I'll take another look to confirm everything is good and then merge this into master for the upcoming release.

Let me know if you have any questions.

Thanks!!!

**Review by fabiolsilva — CHANGES_REQUESTED** [review]

File: csep/utils/readers.py (hunk 1 above)
Comment (fabiolsilva) [reply csep/utils/readers.py:750]: Thanks!

**Review by fabiolsilva — APPROVED** [review]

**Comment by pabloitu** [comment]
reworked into reading gcmt from IRIS instead of ISC, which takes less time and gives less query issues.

**Comment by pabloitu** [comment]
*note:  In the tests/build workflow, the conda environment build is taking too long https://github.com/SCECcode/pycsep/blob/c19ce9f1a7a898987c8697440098bc2f51bc9088/.github/workflows/build-test.yml#L32

and sometimes it gives up of building the environment and just fail the workflow. We maybe should build the environment differently.

**Pull request merged**